objects/f42a3e/contained/a15998/script.lua

```diff
@@ -0,0 +1,418 @@
+	board.reload()
+    setupCoordsUnnamed, setupPiecesUnnamed, setupCoordsNamed, setupPiecesNamed = {}, {}, {}, {}
+    end
+end
+
+function GetSpawnPositions()
+    local strifeablePieces = {
+      ["Explorer"] = true,
+      ["Town"] = true,
+      ["City"] = true
+    }
+    local boards = getMapTiles()
+
+    while true do
+        local moving = false
+        for _, obj in pairs(boards) do
+            if obj.isSmoothMoving() then
+                moving = true
+                break
+            end
+        end
+        if not moving then break end
+        coroutine.yield()
+    end
+
+    local output = {}
+    for _,board in pairs(boards) do
+        local boardTable = {}
+        local hits = Physics.cast({
+            origin = board.getPosition() + Vector(0, 0.45, 0),
+            direction = Vector(0, 1, 0),
+            type = 3,
+            size = board.getBounds().size,
+        })
+        for _,hit in pairs(hits) do
+            if not isIslandBoard({obj=hit.hit_object}) then
+                local subHits = Physics.cast({
+                    origin = hit.hit_object.getPosition() + Vector(0, 0.1, 0),
+                    direction = Vector(0, -1, 0),
+                    max_distance = 0.6,
+                })
+                local onBoard = false
+                for _,subHit in pairs(subHits) do
+                    if subHit.hit_object == board then
+                        onBoard = true
+                        break
+                    end
+                end
+                if onBoard then
+		    local state = hit.hit_object.getStateId()
+                    if state == -1 then
+                        state = 1
+                    end
+                    local name = hit.hit_object.getName()
+                    if strifeablePieces[name] then
+                        local strifeHits = Physics.cast({
+                        origin = hit.hit_object.getPosition(),
+                        direction = Vector(0, 1, 0),
+                        max_distance = 5,
+                        --debug = true
+                        })
+                        for _,strifeHit in pairs(strifeHits) do
+                            if strifeHit.hit_object.getName() == "Strife" then
+                                name = name.."S"
+                                state = strifeHit.hit_object.getStateId()
+                                destroyObject(strifeHit.hit_object)
+                              break
+                          end
+                       end
+                    end
+                    local pos = board.positionToLocal(hit.hit_object.getPosition())
+                    pos.y = 0.7
+                    table.insert(boardTable, {state = state, name = name, position = pos})
+		    destroyObject(hit.hit_object)
+                end
+            end
+        end
+        output[board.getGUID()] = boardTable
+    end
+    return output
+end
+
+function isIslandBoard(params)
```

Comment:
you should use the global functions `isIslandBoard`, `isIsland`, and `getMapTiles()` instead

Reply:
the global getMapTiles() won't work since this isn't taking place on the ocean tile.